sphinx_gallery/gen_rst.py

```diff
@@ -231,6 +235,19 @@ def text2string(content):
         return content
 
 
+def extract_thumbnail_number(text):
+    """ Pull out the thumbnail image number specified in the docstring. """
+
+    # check whether the user has specified a specific thumbnail image
+    pattr = re.compile("^\s*#\s*sphinx_gallery_thumbnail_number\s*=\s*([0-9]+)\s*$", flags=re.MULTILINE)
+    match = pattr.search(text)
+    try:
+        thumbnail_number = int(match.groups()[0])
+    except AttributeError:
+        thumbnail_number = 1 # by default, use the first figure created
+
+    return thumbnail_number
```

Comment:
I would rather write this function as `extract_example_configuration` and let it return a dictionary or named tuple. Then move all the config dictionary declaration inside here so that `split_code_and_text_blocks` remains cleaner. Also I would rename `config` to `example_config` or similar just to give it a bit more contex when reading the source code.